addons/l10n_gr_edi/models/account_move.py

```diff
@@ -0,0 +1,718 @@
+    _inherit = 'account.move'
+
+    l10n_gr_edi_mark = fields.Char(
+        string='MyDATA Mark',
+        compute='_compute_from_l10n_gr_edi_document_ids',
+        store=True,
+    )
+    l10n_gr_edi_cls_mark = fields.Char(
+        string='MyDATA Classification Mark',
+        compute='_compute_from_l10n_gr_edi_document_ids',
+        store=True,
+    )
+    l10n_gr_edi_document_ids = fields.One2many(
+        comodel_name='l10n_gr_edi.document',
+        inverse_name='move_id',
+        copy=False,
+        readonly=True,
+    )
+    l10n_gr_edi_state = fields.Selection(
+        selection=[
+            ('invoice_sent', 'Invoice sent'),
+            ('invoice_error', 'Invoice send failed'),
+            ('bill_fetched', "Expense classification ready to send"),
+            ('bill_sent', "Expense classification sent"),
+            ('bill_error', "Expense classification send failed"),
+        ],
+        string='MyDATA Status',
+        compute='_compute_from_l10n_gr_edi_document_ids',
+        store=True,
+        tracking=True,
+    )
+    l10n_gr_edi_available_inv_type = fields.Char(compute='_compute_l10n_gr_edi_available_inv_type')
+    l10n_gr_edi_correlation_id = fields.Many2one(
+        comodel_name='account.move',
+        string='MyDATA Correlated Invoice',
+    )
+    l10n_gr_edi_inv_type = fields.Selection(
+        selection=INVOICE_TYPES_SELECTION,
+        string='MyDATA Invoice Type',
+        compute='_compute_l10n_gr_edi_inv_type',
+        store=True,
+    )
+    l10n_gr_edi_payment_method = fields.Selection(
+        selection=PAYMENT_METHOD_SELECTION,
+        string='MyDATA Payment Method',
+        default='1',
+    )
+    l10n_gr_edi_warnings = fields.Json(compute='_compute_l10n_gr_edi_warnings')
+    l10n_gr_edi_need_correlated = fields.Boolean(compute='_compute_l10n_gr_edi_need_fields')
+    l10n_gr_edi_need_payment_method = fields.Boolean(compute='_compute_l10n_gr_edi_need_fields')
+    l10n_gr_edi_enable_send_invoices = fields.Boolean(compute='_compute_l10n_gr_edi_enable_send')
+    l10n_gr_edi_enable_send_expense_classification = fields.Boolean(compute='_compute_l10n_gr_edi_enable_send')
+    l10n_gr_edi_attachment_id = fields.Many2one(comodel_name='ir.attachment')
+
+    ################################################################################
+    # Standard Field Computes
+    ################################################################################
+
+    def unlink(self):
+        self.l10n_gr_edi_document_ids.unlink()
+        return super().unlink()
+
+    @api.depends('l10n_gr_edi_document_ids')
+    def _compute_from_l10n_gr_edi_document_ids(self):
+        self.l10n_gr_edi_state = False
+        self.l10n_gr_edi_mark = False
+        self.l10n_gr_edi_cls_mark = False
+
+        for move in self:
+            if gr_documents := move.l10n_gr_edi_document_ids.sorted():
+                gr_document = gr_documents[0]
+                move.l10n_gr_edi_state = gr_document.state
```

Comment:
I'm not sure we want to see the "error" states on the invoice. It would be good to define a common practice for that because in Mexico, we don't have the "error" states on the business documents